File: python code/highgui/trackbar.py
import cv2

thresholdValue = 200
thresholdType = 3
maxValue = 255
maxType = 4
maxBinaryValue = 255

windowName = "Threshold Demo"
trackbatType = "Type:​\n ​0: Binary​\n​ 1: Binary Inverted​\n ​2: Truncate\n ​3: To Zero​\n​ 4: To Zero Inverted"
trackbarValue = "Value"
# call the function to initialize
# 0: Binary
# 1: Binary Inverted
# 2: Threshold Truncated
# 3: Threshold to zero
# 4: Threshold to inverted

# load image
gray = cv2.imread('../../images/threshold.png', cv2.IMREAD_GRAYSCALE)

# create a window to display results
cv2.namedWindow(windowName, cv2.WINDOW_AUTOSIZE)

def thresholdTypeDemo(*args):
    global thresholdType
    thresholdType = args[0]
    thresh, result = cv2.threshold(gray, thresholdValue, maxBinaryValue, thresholdType)
    cv2.imshow(windowName, result)

def thresholdValueDemo(*args):
    global thresholdValue
    thresholdValue = args[0]
    thresh, result = cv2.threshold(gray, thresholdValue, maxBinaryValue, thresholdType)
    cv2.imshow(windowName, result)

# create trackbar to choos type of threshold
cv2.createTrackbar(trackbatType,
                   windowName,
                   thresholdType,
                   maxType,
                   thresholdTypeDemo)

cv2.createTrackbar(trackbarValue,
                   windowName,
                   thresholdValue,
                   maxValue,
                   thresholdValueDemo)

thresholdTypeDemo(0)

thresholdValueDemo(0)

#  wait until user finishes program
while True:
    c = cv2.waitKey(20)
    if c == 27:
        break

File: c++ code/week1/examples/highgui/highgui/trackbar.cpp
#include <opencv2/highgui/highgui.hpp>
#include <opencv2/imgproc/imgproc.hpp>

#include <string>

int thresholdValue = 150;
int thresholdType = 3;
int const maxValue = 255;
int const maxType = 4;
int const maxBinaryValue = 255;

cv::Mat img, gray, dst;

std::string windowName = "Threshold Demo";
std::string trackbarType = "Type:\n 0: Binary\n 1: Binary Inverted\n 2: Truncate\n 3: To zero\n 4: To Zero Inverted";
std::string trackbarValue = "Value";

void thresholdDemo(int, void*);

int main()
{
	// load image
	img = cv::imread("../../../../../images/threshold.png", cv::IMREAD_COLOR);

	// convert image to gray
	cv::cvtColor(img, gray, cv::COLOR_RGB2GRAY);

	// create a window to display results
	cv::namedWindow(windowName, CV_WINDOW_AUTOSIZE);
	// create trackbar to choose type of threshold
	cv::createTrackbar(trackbarType, windowName, &thresholdType, maxType, thresholdDemo);
	cv::createTrackbar(trackbarValue, windowName, &thresholdValue, maxValue, thresholdDemo);
	// call he function to initialize
	thresholdDemo(0, 0);
	// wait until user finishes program
	while (true) {
		int c;
		c = cv::waitKey(20);
		if (static_cast<char>(c) == 27) break;
	}
	return EXIT_SUCCESS;
}

void thresholdDemo(int, void*) {
	/*
	0: Binary
	1: Binary Inverted
	2: Threshold Truncated
	3: Threshold to Zero
	4: Threshold to Zero Inverted
	*/
	cv::threshold(gray, dst, thresholdValue, maxBinaryValue, thresholdType);
	cv::imshow(windowName, dst);
}